support/globalconfig/network.go

```diff
@@ -31,4 +31,16 @@ func ReconcileNetworkConfig(cfg *configv1.Network, hcp *hyperv1.HostedControlPla
 		cfg.Spec.ExternalIP = globalConfig.Network.Spec.ExternalIP
 		cfg.Spec.ServiceNodePortRange = globalConfig.Network.Spec.ServiceNodePortRange
 	}
+
+	// Without this, the CNOs proxy controller refuses to reconcile the proxy status
+	// The CNO only populates this after MTU probing and it is required for the proxy
+	// controller to populate proxy status. Proxy not being populated makes the MTU
+	// probing fail if there is a proxy. Get out of the deadlock by initially populating
+	// it if unset.
+	if len(cfg.Status.ClusterNetwork) == 0 {
+		cfg.Status.ClusterNetwork = cfg.Spec.ClusterNetwork
+	}
+	if len(cfg.Status.ServiceNetwork) == 0 {
+		cfg.Status.ServiceNetwork = cfg.Spec.ServiceNetwork
```

Comment:
how is it done in regular ocp? This is basically invalidating the cno behaviour with no proxy `The CNO only populates this after MTU probing`, do we have concerns with that?

Reply:
yeah but the CNO can not mtu probe without this set, which is the deadlock this change aims to fix

Reply:
or rather, the proxy status doesn't get set, which prevents the MTU probing for being able to work, because it wants to update a configmap which it won't be able without proxy settings, for which it needs the proxy status